File: Intro to ASP.NET MVC 4/[VB]-Intro to ASP.NET MVC 4/VB/MvcMovie/Controllers/MoviesController.vb
Imports System.Data.Entity

Namespace MvcMovie
    Public Class MoviesController
        Inherits System.Web.Mvc.Controller

        Private db As New MovieDBContext

        '
        ' GET: /Movies/

        Function Index() As ActionResult
            Return View(db.Movies.ToList())
        End Function

        '
        ' GET: /Movies/Details/5

        Function Details(Optional ByVal id As Integer = Nothing) As ActionResult
            Dim movie As Movie = db.Movies.Find(id)
            If IsNothing(movie) Then
                Return HttpNotFound()
            End If
            Return View(movie)
        End Function

        '
        ' GET: /Movies/Create

        Function Create() As ActionResult
            Return View()
        End Function

        '
        ' POST: /Movies/Create

        <HttpPost()> _
        Function Create(ByVal movie As Movie) As ActionResult
            If ModelState.IsValid Then
                db.Movies.Add(movie)
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If

            Return View(movie)
        End Function

        '
        ' GET: /Movies/Edit/5

        Function Edit(Optional ByVal id As Integer = Nothing) As ActionResult
            Dim movie As Movie = db.Movies.Find(id)
            If IsNothing(movie) Then
                Return HttpNotFound()
            End If
            Return View(movie)
        End Function

        '
        ' POST: /Movies/Edit/5

        <HttpPost()> _
        Function Edit(ByVal movie As Movie) As ActionResult
            If ModelState.IsValid Then
                db.Entry(movie).State = EntityState.Modified
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If

            Return View(movie)
        End Function

        '
        ' GET: /Movies/Delete/5

        Function Delete(Optional ByVal id As Integer = Nothing) As ActionResult
            Dim movie As Movie = db.Movies.Find(id)
            If IsNothing(movie) Then
                Return HttpNotFound()
            End If
            Return View(movie)
        End Function

        '
        ' POST: /Movies/Delete/5

        <HttpPost()> _
        <ActionName("Delete")> _
        Function DeleteConfirmed(ByVal id As Integer) As RedirectToRouteResult
            Dim movie As Movie = db.Movies.Find(id)
            db.Movies.Remove(movie)
            db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            db.Dispose()
            MyBase.Dispose(disposing)
        End Sub

        Function SearchIndex(ByVal movieGenre As String, ByVal searchString As String) As ActionResult

            Dim genreList As New List(Of String)

            Dim genreQuery = From d In db.Movies Order By d.Genre Select d.Genre

            genreList.AddRange(genreQuery.Distinct)
            ViewData("MovieGenre") = New SelectList(genreList)

            Dim movies = From m In db.Movies Select m

            If String.IsNullOrEmpty(searchString) = False Then
                movies = movies.Where(Function(s) s.Title.Contains(searchString))
            End If

            If String.IsNullOrEmpty(movieGenre) Then
                Return View(movies)
            Else
                Return View(movies.Where(Function(x) x.Genre = movieGenre))
            End If

        End Function

    End Class
End Namespace

File: Intro to ASP.NET MVC 4/[C#]-Intro to ASP.NET MVC 4/C#/MvcMovie/Controllers/MoviesController.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Entity;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using MvcMovie.Models;

namespace MvcMovie.Controllers
{
    public class MoviesController : Controller
    {
        private MovieDBContext db = new MovieDBContext();

        //
        // GET: /Movies/

public ActionResult Index()
{
    return View(db.Movies.ToList());
}

        //
        // GET: /Movies/Details/5

        public ActionResult Details(int id = 0)
        {
            Movie movie = db.Movies.Find(id);
            if (movie == null)
            {
                return HttpNotFound();
            }
            return View(movie);
        }

        //
        // GET: /Movies/Create

        public ActionResult Create()
        {
            return View();
        }

        //
        // POST: /Movies/Create

        [HttpPost]
        public ActionResult Create(Movie movie)
        {
            if (ModelState.IsValid)
            {
                db.Movies.Add(movie);
                db.SaveChanges();
                return RedirectToAction("Index");
            }

            return View(movie);
        }

//
// GET: /Movies/Edit/5

public ActionResult Edit(int id = 0)
{
    Movie movie = db.Movies.Find(id);
    if (movie == null)
    {
        return HttpNotFound();
    }
    return View(movie);
}

//
// POST: /Movies/Edit/5

[HttpPost]
public ActionResult Edit(Movie movie)
{
    if (ModelState.IsValid)
    {
        db.Entry(movie).State = EntityState.Modified;
        db.SaveChanges();
        return RedirectToAction("Index");
    }
    return View(movie);
}

        //
        // GET: /Movies/Delete/5

        public ActionResult Delete(int id = 0)
        {
            Movie movie = db.Movies.Find(id);
            if (movie == null)
            {
                return HttpNotFound();
            }
            return View(movie);
        }

        //
        // POST: /Movies/Delete/5

        [HttpPost, ActionName("Delete")]
        public ActionResult DeleteConfirmed(int id)
        {
            Movie movie = db.Movies.Find(id);
            db.Movies.Remove(movie);
            db.SaveChanges();
            return RedirectToAction("Index");
        }

        public ActionResult SearchIndex(string movieGenre, string searchString) {
            var GenreLst = new List<string>();

            var GenreQry = from d in db.Movies
                           orderby d.Genre
                           select d.Genre;
            GenreLst.AddRange(GenreQry.Distinct());
            ViewBag.movieGenre = new SelectList(GenreLst);

            var movies = from m in db.Movies
                         select m;

            if (!String.IsNullOrEmpty(searchString)) {
                movies = movies.Where(s => s.Title.Contains(searchString));
            }

            if (string.IsNullOrEmpty(movieGenre))
                return View(movies);
            else {
                return View(movies.Where(x => x.Genre == movieGenre));
            }

        }

        protected override void Dispose(bool disposing)
        {
            db.Dispose();
            base.Dispose(disposing);
        }
    }
}